perun-core/src/main/java/cz/metacentrum/perun/core/impl/PerunRolesLoader.java

```diff
@@ -56,15 +58,42 @@ public void loadPerunRoles(JdbcPerunTemplate jdbc) {
 		}
 	}
 
-	public Map<String, JsonNode> loadPerunPolicies() {
-		Map<String, JsonNode> policies = new HashMap<>();
+	/**
+	 * Load policies from the configuration file as list of PerunPolicies
+	 *
+	 * @return list of PerunPolicies
+	 */
+	public List<PerunPolicy> loadPerunPolicies() {
+		List<PerunPolicy> policies = new ArrayList<>();
 		JsonNode rootNode = loadConfigurationFile();
+		//Fetch all policies from the configuration file
 		JsonNode policiesNode = rootNode.get("perun_policies");
 
+		// For each policy node construct PerunPolicy and add it to the list
 		Iterator<String> policyNames = policiesNode.fieldNames();
 		while(policyNames.hasNext()) {
-			String policyname = policyNames.next();
-			policies.put(policyname, policiesNode.get(policyname));
+			String policyName = policyNames.next();
+			JsonNode policyNode = policiesNode.get(policyName);
+			List<Map<String, String>> perunRoles = new ArrayList<>();
+			JsonNode perunRolesNode = policyNode.get("policy_roles");
+
+			//Field policy_roles is saved as List of maps in the for loop
+			for (JsonNode perunRoleNode : perunRolesNode) {
+				Map<String, String> innerRoleMap = new HashMap<>();
+				Iterator<String> roleArrayKeys = perunRoleNode.fieldNames();
+				while (roleArrayKeys.hasNext()) {
+					String role = roleArrayKeys.next();
+					JsonNode roleObjectNode = perunRoleNode.get(role);
+					String object = roleObjectNode.isNull() ? null : roleObjectNode.textValue();
+					innerRoleMap.put(role, object);
+				}
+				perunRoles.add(innerRoleMap);
+			}
+
+			//Field include_policies is saved as List of Strings.
+			List<String> includePolicies = new ArrayList<>(new ObjectMapper().convertValue(policyNode.get("include_policies"), new TypeReference<List<String>>() {}));
```

Comment:
As @zlamalp mentioned, the `ObjectMapper` can be reused as a static member.

Reply:
fixed